### PR TITLE
Balance and bug fixes

### PR DESCRIPTION
**Pull request by itsjustbones**
Lot of balance changes, new upgrade, and bug fixes:

bug fixes:
change supply inadequate message for morphling to tyrannozor for clarity
tychus EOD shield icon level 4/5 fixed
stukov PotV attack icon level 4/5 fixed
Infested Siege Tanks say they can use deep tunnel, but then deep tunnel is a separate unlock (fixed in client already, not in data)
some supreme reinforcements weren't AP units. (mostly zurvan fight units)
add bio tags for infested mercs
update infested unit races to zerg
fix predator nest not being initially frozen in harvest of screams

balance changes:
Maar is now Heroic
broodlord speed upgrade nerfed  from ~3.3 -> 3
guardian base speed buffed from 1.4 -> 2.25
hunterling stun reduced to 1sec for Heroic
2 hunterlings per buy
hunterling jump cooldown increased 10 - > 12
hunterlings cost changed from 400/150 to 300/150  (original value was bugged, x2 intended)
infested siege tank main gun base range from 18 -> 12
remove infested banshee cloak innate +2 range
for infested siege tank main gun attack, remove anti-baneling ai response
infested marine are now valid infested siege tank ammo
reduce infested diamondback base damage from 20, +10 vs armored -> 10, +8 vs armored
reduce infested diamondack base range 8 -> 6, and snare base range 8 -> 6
remove infested diamondback snare baseline
lower infested diamondbnack slime trail damage from 20 -> 12 per second (has period of .5, so damage numbers are 10 -> 6)
remove original frightful fleshwelder
increased base cost of infested siege tank 200/100 -> 200/125
increased base cost of infested diamondback 225/75 -> 225/100
increased base cost of infested liberator 150/100 -> 150/125

new upgrades:
guardian propellan sacs upgrade: speed increased 2.25 -> 3.25
guardian primodial fury upgrade: +10% AS per attack, up to +50% 5s duration.
guardian explosive spores upgrade: adds ability to target ground units with a delayed explosion, aoe attack.  20 dmg, knockback, 30s cooldown.  has autocast
construction jump jets: scv cliff jump upgrade
demolition armaments: widow mine building attack upgrade.  affects death explosion upgrade and splash damage.  adds toggle to disable auto casting on buildings
infested siege tank seismic sonar upgrade:  range increase on main gun +3, and +1 on tentacle
infested banshee advanced targeting optics upgrade: +2 range while cloaked
infested siege tank balanced roots upgrade: attack while moving for infested siege tank tentacle
infested diamondback concentrated spew upgrade: increase attack and snare range by +2
infested diamondback fungal snare is now progressive, 2 levels, first adds snare, second reduces cooldown by 15s.  old improved fungal snare upgrade folded into the progressive one.
infested diamondback violent enzymes upgrade: increase base damage by +8
frightful fleshwelder infested siege tank reduces cost to 200/100
frightful fleshwelder infested diamondback reduces cost to 175/100
frightful fleshwelder infested banshee reduces to cost 125/100
frightful fleshwelder infested liberator reduces to cost 150/100
zerg static defenses excavating claws upgrade: move faster, especially off creep, and root faster

Here videos of some of the new stuff stuff that's got major visual components

guardian cast ability

https://github.com/user-attachments/assets/17aba9b0-9d17-4168-9419-d009d279e20d

widow mine upgrade

https://github.com/user-attachments/assets/53a34127-1b28-46c9-9daa-013eb59e9718

scv cliff jump
https://github.com/user-attachments/assets/5b57c64a-a090-4c92-9c30-2a5096e3337d

### CODE REVIEW

File: Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/UpgradeData.xml [hunk 1]
```diff
@@ -15860,4 +15831,90 @@
...
+        <EditorCategories value="Race:Zerg"/>
+    </CUpgrade>
+    <CUpgrade id="AP_GuardianPrimordialFury">
+        <Icon value="Assets\Textures\btn-ability-zerg-dehaka-guardian-primordialfury.dds"/>
+        <EditorCategories value="Race:Zerg"/>
+    </CUpgrade>
+    <CUpgrade id="AP_SCVCliffJump">
+        <EffectArray Operation="Set" Reference="Unit,AP_SCV,Mover" Value="CliffJumper"/>
+        <EditorCategories value="Race:Terran,UpgradeType:Talents"/>
+        <AffectedUnitArray value="AP_SCV"/>
```
Comment (Ziktofel) [inline Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/UpgradeData.xml:15849]: Add infested SCV as they share their upgrades

File: Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/UpgradeData.xml [hunk 2]
```diff
@@ -15860,4 +15831,90 @@
...
+        <EffectArray Operation="Subtract" Reference="Abil,AP_SpineCrawlerRoot,InfoArray[0].SectionArray[Stats].DurationArray[Delay]" Value="3.000000"/>
+        <EffectArray Operation="Subtract" Reference="Abil,AP_SporeCrawlerRoot,InfoArray[0].SectionArray[Abils].DurationArray[Delay]" Value="3.000000"/>
+        <EffectArray Operation="Subtract" Reference="Abil,AP_SporeCrawlerRoot,InfoArray[0].SectionArray[Actor].DurationArray[Duration]" Value="3.000000"/>
+        <EffectArray Operation="Subtract" Reference="Abil,AP_SporeCrawlerRoot,InfoArray[0].SectionArray[Mover].DurationArray[Delay]" Value="3.000000"/>
+        <EffectArray Operation="Subtract" Reference="Abil,AP_SporeCrawlerRoot,InfoArray[0].SectionArray[Stats].DurationArray[Delay]" Value="3.000000"/>
+        <EffectArray Reference="Unit,AP_SpineCrawlerUprooted,Speed" Value="1"/>
+        <EffectArray Reference="Unit,AP_SporeCrawlerUprooted,Speed" Value="1"/>
+        <EffectArray Operation="Set" Reference="Unit,AP_SpineCrawlerUprooted,SpeedMultiplierCreep" Value="1.4"/>
+        <EffectArray Operation="Set" Reference="Unit,AP_SporeCrawlerUprooted,SpeedMultiplierCreep" Value="1.4"/>
```
Comment (Ziktofel) [inline Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/UpgradeData.xml:15890]: Infested buildings?

Comment (itsjustbones) [reply Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/UpgradeData.xml:15890]: added

File: Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/UpgradeData.xml (hunk 1 above)
Comment (itsjustbones) [reply Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/UpgradeData.xml]: added

File: Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/UpgradeData.xml (hunk 2 above)
Comment (itsjustbones) [reply Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/UpgradeData.xml:15890]: The upgrade specifically targets defensive structures, not all structures.  I think its fine to exclude the production building items, similar to creep stomach.  Envy plans to add creep stomach and this upgrade to infested bunker themselves in a separate pr soon.

**Pull request merged**